RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc

```diff
@@ -166,42 +167,45 @@ namespace {
 	if (ret==-1.f) return ret;
       }
 
+
       // parametrized dz and dr cut by using their error
       if (dz_par1[2]<std::numeric_limits<float>::max() || dr_par1[2]<std::numeric_limits<float>::max()) {
+
 	float maxDz_par1[3];
 	float maxDr_par1[3];
-	dzCut_par1(trk,nLayers,dz_par1, maxDz_par1);
-	drCut_par1(trk,nLayers,dr_par1, maxDr_par1);
-
-        Point bestVertex = getBestVertex(trk,vertices,minNVtxTrk); // min number of tracks 3
-	if (bestVertex.z() < -99998.) {
-	  bestVertex = beamSpot.position();
+	dzCut_par1(trk,nLayers,dz_par1,dz_exp, maxDz_par1);
+	drCut_par1(trk,nLayers,dr_par1,dr_exp, maxDr_par1);
+
+	float maxDz_par[3];
+	float maxDr_par[3];
+	std::copy(std::begin(maxDz_par1),std::end(maxDz_par1),std::begin(maxDz_par));
+	std::copy(std::begin(maxDr_par1),std::end(maxDr_par1),std::begin(maxDr_par));
+
+	// parametrized dz and dr cut by using d0 and z0 resolution
+	if (dz_par2[2]<std::numeric_limits<float>::max() || dr_par2[2]<std::numeric_limits<float>::max()) {      
```

Comment:
I suppose, you will never fall in the situation that par2 values are defined, while par1 are at the ::max values and are disabled. (in case par1 is at max and par2 are defined, the par2 will not be applied because they are inside the "if" over par1).
Please confirm that this is your intention.